### PR TITLE
Refactor TorService

### PR DESCRIPTION
**Pull request by junderw**
This will make it easier to add the extra config elements,

I think we should make the config like:
`BTCPAY_TOR_SERVICES=serviceName1,onionHost1,virtualPort1:serviceName2,onionHost2,virtualPort2`
So it can be parsed by splitting by `:` then splitting by `,` and making sure each service has 3 elements and the 3rd element should be parsed from string to int.

This will make it easier to pass values for node management software that makes use of docker-compose.

See https://github.com/getumbrel/umbrel/issues/607

@Kukks I think this is a first step to what we were talking about.

### CODE REVIEW

**Comment by Kukks** [comment]
I've pushed directly to this PR and added the functionality requested. Slightly different syntax than what you requested but  we already use this format for non-onion related tor services

**Comment by junderw** [comment]
LGTM logic-wise

File: BTCPayServer/Services/TorServices.cs [hunk 1]
```diff
@@ -2,110 +2,157 @@
...
-            network = _networks.GetNetwork<BTCPayNetworkBase>(splitted[0]);
-            return network != null;
+        }
+
+        internal override Task[] InitializeTasks()
+        {
+            return new[] {CreateLoopTask(Refresh)};
+        }
+
+        private Task LoadFromConfig()
```
Comment (NicolasDorier) [inline BTCPayServer/Services/TorServices.cs:145]: 1. Why `Task`
2. Why ignoring `torcservices` if `torrcfile` is set?

**Comment by NicolasDorier** [comment]
The `torservices` should be documented in the `DefaultConfiguration` and be parsed into the BTCPayServerOptions as well.

**Comment by rockstardev** [comment]
@junderw did you allow maintainers to add commits to your PR (see screenshot for setting)?

**Comment by junderw** [comment]
yes, that is why Kukks was able to push commits.

**Comment by Kukks** [comment]
> yes, that is why Kukks was able to push commits.

Are you sure it's not because we are the same person with split personalities that activate based on timezone?

File: BTCPayServer/Services/TorServices.cs (hunk 1 above)
Comment (Kukks) [reply BTCPayServer/Services/TorServices.cs:145]: Removed task, was leftover from before I refactored.

Comment (Kukks) [reply BTCPayServer/Services/TorServices.cs:145]: I ignore since they  might define configs that conflict with one another

**Comment by junderw** [comment]
> Are you sure it's not because we are the same person with split personalities that activate based on timezone?

Or it could be your ssh key in my authorized_keys :thinking:

**Comment by rockstardev** [comment]
OK, it was my bad... I've set everything up and pushed my commits @junderw & @Kukks 

Watch it all happen on: https://www.youtube.com/watch?v=R3WxF1NQsOU ~43 mins

**Comment by junderw** [comment]
LGTM

utACK 5b9dac5c455ef7e01b1ebb2a3717669e6bc811b3

**Comment by NicolasDorier** [comment]
Can we just add that the new config is mutually exclusive with `torrcfile` in the doc? And throw a proper exception if both are specified?

**Comment by junderw** [comment]
Is the documentation addition a part of this repo? I couldn't figure out where the doc change needs to be made.

**Comment by junderw** [comment]
test failures unrelated.

**Comment by Kukks** [comment]
@NicolasDorier  https://www.youtube.com/watch?v=52vsExtoeC0

**Comment by NicolasDorier** [comment]
wolololo

**Pull request merged**